addons/event_sale/models/event_ticket.py

```diff
@@ -56,9 +56,14 @@ def _compute_price_reduce(self):
             # TODO drop price field usage
             # and ideally remove this whole price_reduce logic
             # seems strange to not apply pricelist logic but still use pricelist discount...
-            discount = (
-                product.lst_price - product._get_contextual_price()
-            ) / product.lst_price if product.lst_price else 0.0
+            pricelist = self.env['product.pricelist'].browse(self._context.get('pricelist'))
+            lst_price = product.currency_id._convert(
+                product.lst_price,
+                pricelist.currency_id,
+                self.env.company,
+                fields.Datetime.now()
+            )
+            discount = (lst_price - product._get_contextual_price()) / lst_price if lst_price else 0.0
```

Comment:
there is an issue introduced here:

comparing the `product.lst_price` in one currency and the `product._get_contextual_price()` in another results in a negative discount which resulted in the effect of "currency conversion" in the next line. now with this fix, this crude hack is overruled and if in sale order you change the pricelist to another one with different currency, the ticket price will not be converted.  it also cancels out any discounts set on the sale order line.

a possible fix to both cases is not to use product price at all and use `ticket.price` in the pricelist calculation and find a workaround `_get_contextual_price()` that uses the same price (even copying over its logic here)

@awa-odoo @yosa-odoo @Feyensv @nle-odoo 

another ticket for reference: https://www.odoo.com/web?debug#id=2916904&cids=1&menu_id=4720&action=333&active_id=49&model=project.task&view_type=form

Reply:
We were comparing amounts in different currencies, which was simply wrong ...  Maybe it's highlighting another problem that should be fixed :thinking:.

> results in a negative discount which resulted in the effect of "currency conversion" in the next line
Can you give a clear example ? I'm not sure I understand what you mean here.

>  if in sale order you change the pricelist to another one with different currency, the ticket price will not be converted
We do not want to automatically recompute all lines when you change the pricelist, that's expected, but maybe there is a fix to do (or backport) for the SOL price computation in `event_sale` when you trigger the 'Update prices' action.